src/components/App.jsx

```diff
@@ -0,0 +1,209 @@
+import "../blocks/App.css";
+import { useState, useEffect } from "react";
+import { Route, Routes, Navigate, useNavigate } from "react-router-dom";
+import { SearchContext } from "../contexts/SearchContext";
+import { UserContext } from "../contexts/UserContext";
+import { AppContext } from "../contexts/AppContext";
+import Api from "../utils/api";
+import NewsApi from "../utils/newsApi";
+import { authorize } from "../utils/auth";
+import Nav from "./Nav";
+import Main from "./Main";
+import Footer from "./Footer";
+import ProtectedRoute from "./ProtectedRoute";
+import SavedNews from "./SavedNews";
+import SigninModal from "./SigninModal";
+import SignupModal from "./SignupModal";
+
+export default function App() {
+  const [searchState, setSearchState] = useState({
+    results: [],
+    keyword: "",
+    articlesAvail: 0,
+    articlesShown: 3,
+    loading: false,
+    nothingFound: false,
+  });
+  const [userState, setUserState] = useState({
+    loggedIn: false,
+    email: "",
+    username: "",
+    savedNews: [],
+  });
+  const [activeModal, setActiveModal] = useState("");
+  const [protectedDestination, setProtectedDestination] = useState("");
+  const navigate = useNavigate();
+
+  const api = new Api();
+
+  const newsApi = new NewsApi({
+    baseUrl: "https://nomoreparties.co/news/v2/everything",
+    apiKey: "a16de474931b4e5a83f83ad53ba3df69",
+  });
+
+  useEffect(() => {
+    if (protectedDestination !== "") setActiveModal("signin");
+  }, [protectedDestination]);
+
+  useEffect(() => {
+    document.addEventListener("keydown", (e) => {
+      if (e.key === "Escape") closeActiveModal();
+    });
+
+    document.querySelectorAll(".modal").forEach((modal) => {
```

Comment:
`document.querySelector` or `querySelectorAll` should **not be used** in `React` because it’s a **very slow** operation for the browser. React was specially created not to search anything in the DOM. So, you can directly operate with the markup inside the components. Or you can use hook `useRef` for the reference.


But here you should  add it to the `document` and remove from `document` in the `clean up` function

```js

    const handleOverlay = (e) => {
      if (e.target.classList[0] === "modal") closeActiveModal();
    };

    document.addEventListener("click", handleOverlay);
```

Reply:
moved closing logic directly into modal's "onClick" attribute